sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -716,18 +647,26 @@ export class EventHubSender extends LinkEntity {
             this.name
           );
 
-          this._sender!.on(SenderEvents.accepted, onAccepted);
-          this._sender!.on(SenderEvents.rejected, onRejected);
-          this._sender!.on(SenderEvents.modified, onModified);
-          this._sender!.on(SenderEvents.released, onReleased);
-
-          const delivery = this._sender!.send(message, undefined, 0x80013700);
-          log.sender(
-            "[%s] Sender '%s', sent message with delivery id: %d",
-            this._context.connectionId,
-            this.name,
-            delivery.id
-          );
+          try {
+            const delivery = await this._sender!.send(message, undefined, 0x80013700);
+            log.sender(
+              "[%s] Sender '%s', sent message with delivery id: %d",
+              this._context.connectionId,
+              this.name,
+              delivery.id
+            );
+          } catch (err) {
+            err = translate(err);
+            log.error(
+              "[%s] An error occurred while sending the message",
+              this._context.connectionId,
+              err
+            );
+            return reject(err);
+          } finally {
+            clearTimeout(waitTimer);
```

Comment:
I think you'd want to call `removeListeners` here to also remove the `onAbort` function from the `abortSignal`.

Reply:
Agree, fixed